src/client/options/test.rs

```diff
@@ -332,3 +332,43 @@ async fn run_uri_options_spec_tests() {
 async fn run_connection_string_spec_tests() {
     run_spec_test(&["connection-string"], run_test).await;
 }
+
+async fn parse_uri(option: &str, suggestion: Option<&str>) {
+    match ClientOptionsParser::parse(&format!("mongodb://host:27017/?{}=test", option))
+        .as_ref()
+        .map_err(|e| e.as_ref())
+    {
+        Ok(_) => panic!("expected error for option {}", option),
+        Err(ErrorKind::ArgumentError { message, .. }) => {
+            match suggestion {
+                Some(s) => assert_eq!(
+                    message,
+                    &format!(
+                        "{} is an invalid option. An option with a similar name exists: {}",
+                        option, s
+                    )
+                ),
+                None => assert_eq!(message, &format!("{} is an invalid option", option)),
+            };
+        }
+        Err(e) => panic!("expected ArgumentError, but got {:?}", e),
+    }
+}
+
+#[cfg_attr(feature = "tokio-runtime", tokio::test)]
+#[cfg_attr(feature = "async-std-runtime", async_std::test)]
+async fn parse_unknown_options() {
```

Comment:
I like these tests! The ones that give suggestions are all really good examples of things users might accidentally put in their URIs.